File: backend/app/schemas/audit.py
import uuid
from datetime import datetime
from pydantic import BaseModel


class AuditLogItem(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID | None
    username: str | None
    action: str
    resource_type: str
    resource_id: str
    detail: dict
    ip_address: str | None
    created_at: datetime


class AuditLogListResponse(BaseModel):
    items: list[AuditLogItem]
    total: int

File: frontend/src/types/audit.ts
export interface AuditLogItem {
  id: string
  user_id: string | null
  username: string | null
  action: string
  resource_type: string
  resource_id: string
  detail: Record<string, unknown>
  ip_address: string | null
  created_at: string
}

export interface AuditLogListResponse {
  items: AuditLogItem[]
  total: number
}

export interface AuditLogFilters {
  action?: string
  resource_type?: string
  start_date?: string
  end_date?: string
}
